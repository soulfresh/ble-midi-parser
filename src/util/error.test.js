/* eslint-env jest */
/* eslint no-console: 0 */
test('error(...args)', () => {
  const consoleError = console.error;
  console.error = jest.fn();
  const error = require('./error').default;
  error();
  error(new Error('MOCK_ERROR_MSG_1'));
  error('MOCK_ERROR_MSG_2');
  error('MOCK_ERROR_MSG_3', 'MOCK_ERROR_MSG_4');
  expect(console.error.mock).toMatchSnapshot();
  console.error = consoleError;
});
