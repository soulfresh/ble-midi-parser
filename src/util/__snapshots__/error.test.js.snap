// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error(...args) 1`] = `
Object {
  "calls": Array [
    Array [],
    Array [
      [Error: MOCK_ERROR_MSG_1],
    ],
    Array [
      "MOCK_ERROR_MSG_2",
    ],
    Array [
      "MOCK_ERROR_MSG_3",
      "MOCK_ERROR_MSG_4",
    ],
  ],
  "instances": Array [
    BufferedConsole {
      "_buffer": Array [],
      "assert": [Function],
      "clear": [Function],
      "count": [Function],
      "countReset": [Function],
      "debug": [Function],
      "dir": [Function],
      "dirxml": [Function],
      "error": [Function],
      "group": [Function],
      "groupCollapsed": [Function],
      "groupEnd": [Function],
      "info": [Function],
      "log": [Function],
      "time": [Function],
      "timeEnd": [Function],
      "trace": [Function],
      "warn": [Function],
      Symbol(counts): Map {},
      Symbol(groupIndent): "",
    },
    BufferedConsole {
      "_buffer": Array [],
      "assert": [Function],
      "clear": [Function],
      "count": [Function],
      "countReset": [Function],
      "debug": [Function],
      "dir": [Function],
      "dirxml": [Function],
      "error": [Function],
      "group": [Function],
      "groupCollapsed": [Function],
      "groupEnd": [Function],
      "info": [Function],
      "log": [Function],
      "time": [Function],
      "timeEnd": [Function],
      "trace": [Function],
      "warn": [Function],
      Symbol(counts): Map {},
      Symbol(groupIndent): "",
    },
    BufferedConsole {
      "_buffer": Array [],
      "assert": [Function],
      "clear": [Function],
      "count": [Function],
      "countReset": [Function],
      "debug": [Function],
      "dir": [Function],
      "dirxml": [Function],
      "error": [Function],
      "group": [Function],
      "groupCollapsed": [Function],
      "groupEnd": [Function],
      "info": [Function],
      "log": [Function],
      "time": [Function],
      "timeEnd": [Function],
      "trace": [Function],
      "warn": [Function],
      Symbol(counts): Map {},
      Symbol(groupIndent): "",
    },
    BufferedConsole {
      "_buffer": Array [],
      "assert": [Function],
      "clear": [Function],
      "count": [Function],
      "countReset": [Function],
      "debug": [Function],
      "dir": [Function],
      "dirxml": [Function],
      "error": [Function],
      "group": [Function],
      "groupCollapsed": [Function],
      "groupEnd": [Function],
      "info": [Function],
      "log": [Function],
      "time": [Function],
      "timeEnd": [Function],
      "trace": [Function],
      "warn": [Function],
      Symbol(counts): Map {},
      Symbol(groupIndent): "",
    },
  ],
}
`;
